addons/point_of_sale/migrations/8.0.1.0.1/post-migration.py

```diff
@@ -20,14 +20,103 @@
 #
 ##############################################################################
 
+import logging
 from openerp.openupgrade import openupgrade
+from openerp.modules.registry import RegistryManager
+from openerp import SUPERUSER_ID
+
+
+logger = logging.getLogger('OpenUpgrade.point_of_sale')
+
+
+def get_warehouse_id(cr, pool, shop_id):
+    """
+        Get the sale.shop warehouse_id
+    """
+    cr.execute(
+        "SELECT warehouse_id FROM sale_shop WHERE id=%s", (shop_id,))
+    return cr.fetchone()[0]
+
+
+def get_stock_location_id(cr, pool, shop_id):
+    """
+        Select the stock_location_id (internal) on pos.config
+        to be that of the stock_warehouse of the old shop_id
+    """
+    wh_obj = pool['stock.warehouse']
+    return wh_obj.read(cr, SUPERUSER_ID,
+        [get_warehouse_id(cr, pool, shop_id)],
+        ['lot_stock_id'])[0]['lot_stock_id'][0]
+
+
+def get_company_id(cr, pool, shop_id, pc):
+    """
+        Try to get company_id from sale.shop
+        or set it to any res.company record
+    """
+    cr.execute(
+        "SELECT company_id FROM sale_shop WHERE id=%s", (shop_id,))
+    comp_id = cr.fetchone()
+    if not comp_id:
+        comp_obj = pool['res.company']
+        logger.warning(
+            "Could not determine exactly company_id for pos.config with id=%s (%s)." % (pc.id, pc.name))
+        comp_id = comp_obj.search(cr, SUPERUSER_ID, [])
+    return comp_id[0]
```

Comment:
I've set it randomly because company_id is a required field on pos.config and thought leaving it null could cause an issue. I see your point but in any case Admin user can change the value later if he determines it is not OK.
Maybe warning message could be more informative.